store/src/test/java/discord4j/store/service/StoreTests.java

```diff
@@ -57,6 +59,63 @@ public void testGenericStore() {
         new StoreTest(Objects.requireNonNull(store)).test();
     }
 
+    @Test
+    public void testQuery() {
```

Comment:
This helps understanding how to leverage the API but I'm having trouble seeing how can we use this to implement missing features, specifically querying to stores that have composite keys, or is it only for querying fields on values?

Reply:
With this system, composite keys won't be necessary as this provides a way of querying arbitrary fields on values. Take the message store for example, on a guild delete we cannot currently clear messages from this guild without doing a horrible mess of code.

Reply:
Or, O(n) complexity (iterating and checking guild id). Which, may not be necessarily bad considering it could be parallelized.

Reply:
Not necessarily though. Take sqlite for example, it only allows a single active connection as it is file-based making parallelization impossible or slower than sequential execution

Reply:
You can query an entire table, parallelize through looking for matching guild IDs (as this happens in the JVM), then bulk remove.

Reply:
That still doesn't sound very appealing though, especially with the existence of this PR.